src/map/clif.cpp

```diff
@@ -2254,11 +2258,9 @@ void clif_selllist(map_session_data *sd)
 	if (!sd->npc_shopid || (nd = map_id2nd(sd->npc_shopid)) == NULL)
```

Comment:
Variable declarations should be fixed.
fd should be removed.
i should be moved into the for loop declaration.
A TODO should be added to convert sd to reference and the nullpo_ret should be changed to a normal nullpointer check